.github/workflows/test.yml

```diff
@@ -49,10 +49,75 @@ jobs:
           DANGER_GITHUB_API_TOKEN: ${{ secrets.DANGER_GITHUB_API_TOKEN }}
         run: bundle exec danger
 
+  ios_test:
+    name: Running tests in iOS
+    runs-on: macos-latest
+    defaults:
+      run:
+        working-directory: ./sample/ios
+    steps:
+      - uses: actions/checkout@v3
+      - uses: ruby/setup-ruby@v1
+        with:
+          ruby-version: '3.2'
+          bundler-cache: true
+
+      - name: Setup Java JDK
+        uses: actions/setup-java@v2.1.0
+        with:
+          distribution: 'adopt'
+          java-version: '17'
+
+      - name: Remove the old sample project
+        run: rm -rf sample
+        working-directory: ./
+
+      - name: Generate the new sample project
+        run: sh make.sh --bundle-id co.nimble.kmm.sample --bundle-id-staging co.nimble.kmm.sample.staging --project-name sample --ios-version 14.0
+        working-directory: ./
+
+      - name: Bundle install
+        run: bundle install --path vendor/bundle
+
+      - name: Cache Pods
+        uses: actions/cache@v3
+        id: cocoapodCache
+        with:
+          path: Pods
+          key: ${{ runner.os }}-pods-${{ hashFiles('**/Podfile.lock') }}
+          restore-keys: |
+            ${{ runner.os }}-pods-
+
+      - name: Generate KMM frameworks for Cocoapods
+        run: |
+          cd ..
+          ./gradlew generateDummyFramework
+
+      - name: Install Pods Dependencies
+        run: bundle exec pod install
+
+      - name: Build and Test
+        run: bundle exec fastlane buildAndTest
+        env:
+          CI: true
+
+      - name: Clean up previous code coverage report
+        run: bundle exec fastlane cleanUpOutput
+
+      - name: Review pull request by Danger
+        uses: MeilCli/danger-action@v5
+        with:
+          plugins_file: 'sample/ios/Gemfile'
+          install_path: 'sample/ios/vendor/bundle'
+          danger_file: 'sample/ios/Dangerfile'
+          danger_id: 'danger-pr'
+        env:
+          DANGER_GITHUB_API_TOKEN: ${{ secrets.GITHUB_TOKEN }}
```

Comment:
I don't think the `working-directory` setting from the workflow could be applied for this action 🤔 Hence, we have to modify the paths in the Dangerfile as below to make it work.

How about coming back to use the normal run action instead of this action to leverage the `working-directory` setting? With a `working-directory` set, this run action will grab the Gemfile and the Dangerfile from that custom dir and all paths in the Dangerfile will be correct to start from the custom dir 💭 

```suggestion
      - name: Run Danger to wrap up the review
        env:
          DANGER_GITHUB_API_TOKEN: ${{ secrets.GITHUB_TOKEN }}
        run: |
          bundle install --path vendor/bundle
          bundle exec danger
```

Reply:
Please check out my test here https://github.com/nimblehq/kmm-templates/pull/38/files#diff-faff1af3d8ff408964a57b2e475f69a6b7c7b71c9978cccc8f471798caac2c88 🙏 

Log here https://github.com/nimblehq/kmm-templates/actions/runs/5971010851/job/16199704456

Reply:
@luongvo I think @markgravity is trying to run danger in the iOS project, not the sample project. You guys might need to check again about this.

With Danger in iOS, there should be xcov and SwiftFormat reports ([an example](https://github.com/nimblehq/kmm-templates/actions/runs/5960000787/job/16166591788)).
![image](https://github.com/nimblehq/kmm-templates/assets/19943832/97247859-5910-4d00-a8d8-2ce60a02c585)

So we might need to decide to combine both Dangerfile or not 💭

Reply:
@suho Valid issue cc @phongvhd93 However, I think we should handle this later in a separate task, as most of us execute Danger separately between Android and iOS right now 💭

Reply:
let me try it 👍

Reply:
@markgravity Not sure if you checked my comment, but you should not follow Lucas's approach. It will not run the Danger for iOS project 😔

Reply:
@markgravity cc @luongvo Just checked with Lucas, if the Danger reports are right (having swiftformat and xcov), you can keep the current approach

Reply:
It seems it works as expected. We can see the danger comment 😍